app-spec.yml

```diff
@@ -0,0 +1,64 @@
+alerts:
+  - rule: DEPLOYMENT_FAILED
+  - rule: DOMAIN_FAILED
+envs:
+  - key: POLYGONSCAN_API_KEY
+    scope: RUN_AND_BUILD_TIME
+    type: SECRET
+    value: ${POLYGONSCAN_API_KEY}
+  - key: WEB3_INFURA_PROJECT_ID
+    scope: RUN_AND_BUILD_TIME
+    type: SECRET
+    value: ${WEB3_INFURA_PROJECT_ID}
+name: discord-bots
+region: nyc
+workers:
+  - environment_slug: python
+    envs:
+      - key: DISCORD_BOT_TOKEN_REBASE
+        scope: RUN_AND_BUILD_TIME
+        type: SECRET
+        value: ${DISCORD_BOT_TOKEN_REBASE}
+      - key: DISCORD_REBASE_BOT_WEBHOOK_URL
+        scope: RUN_AND_BUILD_TIME
+        type: SECRET
+        value: ${DISCORD_BOT_WEBHOOK_REBASE}
+    image:
+      registry_type: DOCR
+      repository: discord-bots
+      tag: latest
+    instance_count: 1
+    instance_size_slug: basic-xxs
+    name: next-rebase
+    run_command: python src/next_rebase/main.py
+    source_dir: /
+  - environment_slug: python
+    envs:
+      - key: DISCORD_BOT_TOKEN
+        scope: RUN_AND_BUILD_TIME
+        type: SECRET
+        value: ${DISCORD_BOT_TOKEN_KLIMA_PRICE}
+    image:
+      registry_type: DOCR
+      repository: discord-bots
+      tag: latest
+    instance_count: 1
+    instance_size_slug: basic-xxs
+    name: klima-price
+    run_command: python src/klima_price/main.py
+    source_dir: /
+  - environment_slug: python
+    envs:
+      - key: DISCORD_BOT_TOKEN
+        scope: RUN_AND_BUILD_TIME
+        type: SECRET
+        value: ${DISCORD_BOT_TOKEN_BCT_PRICE}
+    image:
+      registry_type: DOCR
+      repository: discord-bots
+      tag: latest
```

Comment:
Is it possible to use a more meaningful tag based on whatever tag was published in the publish step? Or is there a good reason we're using `latest`?

Using `latest` is typically an antipattern - ideally we'd use the commit hash or something else that is unique so that we aren't constantly republishing the same tag

Reply:
Mainly for simplicity. In the push step, I could tag it with the current commit hash and then use the same tag for pulling. I'll implement that.